src/cycax/cycad/engine.py

```diff
@@ -1,13 +1,217 @@
 import json
-
+import subprocess
+from cycax.cycad.location import TOP, BOTTOM, LEFT, RIGHT, FRONT, BACK
+from cycax.cycad.cycad_part import CycadPart
 
 class Engine:
-    def __init__(self) -> None:
-        pass
+    """
+    This class willbe used to decode a json to a scad file which can be rendered in openscad for 3D view.
+    """
+    dif = 0
+
+    def decode_cube(self, lookup: dict)->str:
+        """
+        This method will return the string that will have the scad for a cube.
+        
+        Args:
+            lookup: this will be the dictionary that contains the details about the cube so that is can be encoded in scad.
+            
+        """
+        center = ""
+        if lookup["center"] == True:
+            center = ", center=true"
+        res = (
+            "cube(["
+            + str(lookup["x_size"])
+            + ", "
+            + str(lookup["y_size"])
+            + ","
+            + str(lookup["z_size"])
+            + "]"
+            + center
+            + ");"
+        )
+        return res
+    
+    def decode_external(self, data_file: str)->str:
+        """
+        This method will return the scad string necessary for processing the external part.
+        
+        Args:
+            data_file : This will contain the part number of the external part so that it can be fetched from the parts_stl folder. 
+              
+        """
+        
+        return 'import("../parts_stl/' + data_file + '.stl");'
+
+    def decode_hole(self, lookup: dict)->str:
+        """
+        This method will return the string that will have the scad for a hole.
+        
+        Args: 
+            lookup : This will be a dictionary containing the necessary information about the hole.
+            
+        """
+        tempdiam = lookup["diameter"] / 2
+        res = (
+            "cylinder(r=" + str(tempdiam) + ", h=" + str(lookup["depth"]) + ", $fn=64);"
+        )
+        return res
+
+    def decode_nut(self, lookup: dict)->str:
+        """
+        This method will return the string that will have the scad for a nut cut out.
+        
+        Args:
+            lookup : This will be a dictionary containing the necessary information about the nut.
+            
+        """
+        res = (
+            "rotate(["
+            + str(0)
+            + ", "
+            + str(0)
+            + ", "
+            + str(0)
+            + "])"
+            + "cylinder(r="
+            + str(lookup["nut_type"])
+            + ", h="
+            + str(lookup["depth"])
+            + ", $fn=6);"
+        )
+        return res
+
+    def decode_cut(self)->str:
+        """
+        This method returns a simple scad string neceseray to cut.
```

Comment:
I think we should say **OpenSCAD** since that is the specific language. The file extension is '.scad' but as I understand the format is called OpenSCAD.